marge/single_merge_job.py

```diff
@@ -56,8 +56,8 @@ def update_merge_request_and_accept(self, approvals):
                     merge_request,
                     source_repo_url=source_repo_url,
                 )
-            except GitLabRebaseResultMismatch:
-                log.info("Gitlab rebase didn't give expected result")
+            except GitLabRebaseResultMismatch as err:
+                log.info("Gitlab rebase didn't give expected result: " + err)
```

Comment:
...or should this be `err.reason`? I don't work so much with Python so please help me get this right. The idea is to ensure that the exception message (as given in `synchronize_using_gitlab_rebase`) gets included in the log message here.

Reply:
You can use `err.reason` here, yes. The definition of that class is:

```python
class CannotMerge(Exception):
    @property
    def reason(self):
        args = self.args
        if not args:
            return 'Unknown reason!'

        return args[0]

class GitLabRebaseResultMismatch(CannotMerge):
    def __init__(self, gitlab_sha, expected_sha):
        super().__init__(
            "GitLab rebase ended up with a different commit:"
            "I expected %s but they got %s" % (expected_sha, gitlab_sha)
        )
```

and the `@property` annotation means that you can use `reason` as an attribute.

Reply:
Thanks, incorporated this suggestion now. :bow: